plugin.py

```diff
@@ -404,18 +404,29 @@ def run(self):
 
     def selected_unit_test_method_names(self, view):
         """
-        If all selections are test methods returns an array of all selected
-        test method names; otherwise None
+        Returns an array of selected test method names.
+        Selection can be anywhere inside one or more test methods.
+        If no selection is found inside any test method, then all test method names are returned.
         """
 
-        # @todo should be a scoped selection i.e. is the selection a source.php entity.name.function
         method_names = []
-        for region in view.sel():
-            word = view.substr(view.word(region))
-            if not is_valid_php_identifier(word) or word[:4] != 'test':
-                return None
+        function_areas = view.find_by_selector('meta.function')
+        function_regions = view.find_by_selector('entity.name.function')
 
```

Comment:
If no function areas or regions are found default to the previous implementation because some syntaxes or old syntaxes like my php-grammar plugin don't have those scopes so this feature will fail on those syntaxes.